### DISCUSSION TITLE
Add icon/Image to notification

### DISCUSSION THREAD

**Discussion by leewyatt** [AlmasB/FXGL#1148] [Ideas]
The notification style is now concise and dynamic;

**Notifications** currently only seems to support **String**, 
and the component inside seems to be **Text**.

It would be better to support **Node** or **Label** components.
Because Labeled/Label components can set icons through the **setGraphic** method.

![image](https://user-images.githubusercontent.com/75261429/152481512-248f0576-b3c5-40a8-8e88-379a783959c4.png)

**Comment by AlmasB** [comment]
I like the idea being able to set an icon.

Currently custom implementations can be provided via `notificationViewClass` in the game settings. Hence, `NotificationService` does not know what components are present and only sends raw data, specifically, background color, text color and the `String` message. I'm happy to add another data item - `Node` icon, so the added API will be smth like:

```
getNotificationService().pushNotification(Node icon, String message);
```

It will be up to the implementation of `notificationViewClass` to decide what to do with this extra information. The built-in Xbox style notification will display the icon inside the circle. Let me know if this sounds reasonable.

**Comment by AlmasB** [comment]
@leewyatt now done as part of #1149 

The API is:

```
getNotificationService().pushNotification(message, icon);
```

See [tweet](https://twitter.com/AlmasBaim/status/1549489474488385536) for demo.
